hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/webapp/MockDefaultRequestInterceptorREST.java

```
@@ -830,64 +843,193 @@ public Response listReservation(String queue, String reservationId, long startTi
+      if (resReqsInfo == null || resReqsInfo.getReservationRequest() == null
+          || resReqsInfo.getReservationRequest().size() == 0) {
+        throw new BadRequestException("The ReservationDefinition should " +
+            "contain at least one ReservationRequest");
+      }
+
+      if (resContext.getReservationId() == null) {
+        throw new BadRequestException("Update operations must specify an existing ReservaitonId");
+      }
+
+      ReservationRequestInterpreter[] values = ReservationRequestInterpreter.values();
+      ReservationRequestInterpreter requestInterpreter =
+          values[resReqsInfo.getReservationRequestsInterpreter()];
+      List<ReservationRequest> list = new ArrayList<>();
+
+      for (ReservationRequestInfo resReqInfo : resReqsInfo.getReservationRequest()) {
+        ResourceInfo rInfo = resReqInfo.getCapability();
+        Resource capability = Resource.newInstance(rInfo.getMemorySize(), rInfo.getvCores());
+        int numContainers = resReqInfo.getNumContainers();
+        int minConcurrency = resReqInfo.getMinConcurrency();
+        long duration = resReqInfo.getDuration();
+        ReservationRequest rr =
+            ReservationRequest.newInstance(capability, numContainers, minConcurrency, duration);
+        list.add(rr);
+      }
+
+      ReservationRequests reqs = ReservationRequests.newInstance(list, requestInterpreter);
+      ReservationDefinition rDef = ReservationDefinition.newInstance(
+          resInfo.getArrival(), resInfo.getDeadline(), reqs,
+          resInfo.getReservationName(), resInfo.getRecurrenceExpression(),
+          Priority.newInstance(resInfo.getPriority()));
+      ReservationUpdateRequest request = ReservationUpdateRequest.newInstance(
+          rDef, ReservationId.parseReservationId(resContext.getReservationId()));
+
+      ClientRMService clientService = mockRM.getClientRMService();
+      clientService.updateReservation(request);
+
+    } catch (Exception ex) {
```

This is a fairly broad catch, wouldn't it be better to just throw the exceptions?

---

I agree with you, I will modify the code.